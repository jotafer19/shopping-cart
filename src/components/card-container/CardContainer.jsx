import styles from "./CardContainer.module.css"

export default function CardContainer({ children }) {
    return (
        <div className={styles.container}>
            {children}
        </div>
    )
}