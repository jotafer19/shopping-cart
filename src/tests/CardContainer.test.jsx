import { vi, describe, it, expect } from "vitest"
import { screen, render } from "@testing-library/react"
import CardContainer from "../components/card-container/CardContainer"
import userEvent from "@testing-library/user-event";

vi.mock('../components/item-card/Card', () => ({
    default: ({ item }) => (
        <div data-testId="card">
            <img src={item.image} />
            <div>{item.title}</div>
            <div>{item.rating.rate}</div>
            <div>{item.count}</div>
        </div>
    ),
}));

const data = [
    { id: 1, title: "foo", image: "#", rating: { rate: 3.1 }, count: 1, price: 10, category: "electronics" },
    { id: 2, title: "bar", image: "#", rating: { rate: 4.0 }, count: 5, price: 20, category: "jewelery" }
]

describe("Test CardContainer component", () => {
    it("render component", () => {
        const { container } = render(<CardContainer data={data} />)

        expect(container).toMatchSnapshot()
    })

    it("render category buttons and select options", () => {
        render(<CardContainer data={data} />)

        expect(screen.getByRole("button", { name: "All Products"})).toBeInTheDocument()
        expect(screen.getByRole("button", { name: "Men's Clothing"})).toBeInTheDocument()
        expect(screen.getByRole("button", { name: "Jewelery"})).toBeInTheDocument()
        expect(screen.getByRole("button", { name: "Electronics"})).toBeInTheDocument()
        expect(screen.getByRole("button", { name: "Women's Clothing"})).toBeInTheDocument()
        expect(screen.getByRole("option", { name: "Recommended"})).toBeInTheDocument()
        expect(screen.getByRole("option", { name: "Alphabetical (A-Z)"})).toBeInTheDocument()
        expect(screen.getByRole("option", { name: "Alphabetical (Z-A)"})).toBeInTheDocument()
        expect(screen.getByRole("option", { name: "Price: low to high"})).toBeInTheDocument()
        expect(screen.getByRole("option", { name: "Price: high to low"})).toBeInTheDocument()
    })

    it("check category button click", async () => {
        const user = userEvent.setup()

        render(<CardContainer data={data} />)

        const allProducts = screen.getByRole("button", { name: "All Products"})
        const menProducts = screen.getByRole("button", { name: "Men's Clothing"})
        const jewelery = screen.getByRole("button", { name: "Jewelery"})

        await user.click(jewelery);
        expect(screen.getAllByTestId("card").length).toBe(1)

        await user.click(menProducts)
        expect(screen.queryByTestId("card")).toBeFalsy()

        await user.click(allProducts)
        expect(screen.getAllByTestId("card").length).toBe(2)
    })

    it("change product sort", async () => {
        const user = userEvent.setup()

        render(<CardContainer data={data} />)

        const select = screen.getByRole("combobox")

        await user.selectOptions(select, "low-high")

        expect(select).toHaveValue("low-high")
    })
})