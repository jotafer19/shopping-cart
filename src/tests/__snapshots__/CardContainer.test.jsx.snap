// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test CardContainer component > render component 1`] = `
<div>
  <div
    class="_shop_4ea041"
  >
    <div
      class="_category_4ea041"
    >
      <ul>
        <li>
          <button
            class="_active_4ea041"
          >
            All Products
          </button>
        </li>
        <li>
          <button
            class=""
          >
            Men's Clothing
          </button>
        </li>
        <li>
          <button
            class=""
          >
            Jewelery
          </button>
        </li>
        <li>
          <button
            class=""
          >
            Electronics
          </button>
        </li>
        <li>
          <button
            class=""
          >
            Women's Clothing
          </button>
        </li>
      </ul>
    </div>
    <div
      class="_container_4ea041"
    >
      <div
        class="_filter_4ea041"
      >
        <input
          class="_search_4ea041"
          id="search"
          name="search"
          placeholder="Search products..."
          type="text"
          value=""
        />
        <div
          class="_select_4ea041"
        >
          <label
            for="sort-order"
          >
            Sort by
          </label>
          <select
            id="sort-order"
            name="sort-order"
          >
            <option
              value="recommended"
            >
              Recommended
            </option>
            <option
              value="a-z"
            >
              Alphabetical (A-Z)
            </option>
            <option
              value="z-a"
            >
              Alphabetical (Z-A)
            </option>
            <option
              value="low-high"
            >
              Price: low to high
            </option>
            <option
              value="high-low"
            >
              Price: high to low
            </option>
          </select>
        </div>
      </div>
      <div
        class="_items_4ea041"
      >
        <div
          class="_container_ddcde1 "
        >
          <div
            class="_message_ddcde1"
          >
            Item added to cart!
          </div>
          <div
            class="_bar_ddcde1"
          />
        </div>
        <div
          class="_products_4ea041"
        >
          <div
            data-testid="card"
          >
            <img
              src="#"
            />
            <div>
              foo
            </div>
            <div>
              3.1
            </div>
            <div>
              1
            </div>
          </div>
          <div
            data-testid="card"
          >
            <img
              src="#"
            />
            <div>
              bar
            </div>
            <div>
              4
            </div>
            <div>
              5
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
